### PR TITLE
Fixes to Linux Asset Processing.

### PR DESCRIPTION
**Pull request by Twolewis**
Fixes Linux Asset Processing errors.

don't process materials for "server"
Mark long as a Number type, so that ScriptCanvas "ExtractProperty" node won't fail.
Some debugging prints & errors, and fixing a spelling error.
How was this PR tested?
Locally, on Jenkins

### CODE REVIEW

File: Gems/Gestures/Code/Include/Gestures/GestureRecognizerDragRight.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #pragma once
 
 // Add the DragRight gesture
-#if defined(CARBONATED)
+#if defined(CARBONATED) && 0
```
Comment (Twolewis) [inline Gems/Gestures/Code/Include/Gestures/GestureRecognizerDragRight.h:11]: @aefimov3 - Need your eyes on this.  The "DragRight" gesture caused AP processing errors because the type wasn't reflected.  I tried to reflect it, but it collided with the the "Drag" gesture it inherited from.  I can't see it being used -anywhere- so I disabled it for now.  I'd like to completely delete it, lmk.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.cpp [hunk 2]
```diff
@@ -52,12 +52,8 @@ namespace NodeCpp
         // This change resolves this error by setting the version higher than any previous version.
         FixedVersioningIssue = 14,
 
-#if defined(CARBONATED)
```
Comment (Twolewis) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.cpp:55]: @aefimov3 - I need to understand why this change was made in the first place.  It seems a "Bad Thing(c)" to peg the current version from something prior.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/DataTypeUtils.cpp [hunk 3]
```diff
@@ -335,10 +335,17 @@ namespace ScriptCanvas
 
         bool IsNumber(const AZ::Uuid& type)
         {
+#if defined(CARBONATED)
```
Comment (Twolewis) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/DataTypeUtils.cpp:338]: Added <long> because linux.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.cpp (hunk 2 above)
Comment (aefimov3) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Node.cpp:55]: In LY the node version is 12, which is explicitly set in the code. So we set O3DE's to 12 too on June 20, 2023 to avoid error messages and probably conversion issues.  Default node version in O3DE was actually 2 that time, the constant 14 that changes the default to 15 was inserted later. It looks like they have a valid conversion code piece now, it makes sense to try removing the patch.
